specs/batching.md

```diff
@@ -0,0 +1,59 @@
+# Batch submitting
+
+The process that signs and publishes transactions is not part of the rollup node, but an external “batcher”.
+This is effectively the inverse function of the derivation process, but specified here for posterity. This API design is not enforced by the protocol.
+
+The batcher may use any data-availability platform that registers batch data within a chain of block-hashes and block-numbers.
+
+The rollup node is mostly agnostic to the data platform: as a verifier the channel-bank has to traverse the chain and data has to be written to the bank.
+Ethereum calldata is supported by default, other data-sources may be supported in future upgrades.
+
+Between the batcher and rollup node there exists a simple single-method RPC:
+
+- method: `optimism_batchData`
+- params:
+  1. object, with fields:
+     - `history`: a dictionary of any previous submitted channel IDs (key, `DATA`) and the last frame number (value, `QUANTITY`) that was submitted for these.
+     - `min_size`: `QUANTITY`, the minimum amount of data that may be returned. (TODO)
+     - `max_size`: `QUANTITY`, the maximum amount of data that may be returned. For the batcher to configure, to fit whatever the data host can process at a time.
+     - `max_blocks_per_channel`: `QUANTITY`, the maximum number of blocks to put in the same frame together.
```

Comment:
We went back and forth between doing i per frame or per channel. I think per-channel makes more sense because of compression covering the whole channel